pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1404,6 +1407,66 @@ func doAssignIoAdaptersToDomain(ctx *domainContext, config types.DomainConfig,
 	return nil
 }
 
+func addFileFromCloudInit(file cloudconfig.File, rootPath string) error {
+	// transform file.Permission to os.FileMode
+	perm, err := strconv.ParseUint(file.RawFilePermissions, 8, 32)
+	if err != nil {
+		return err
+	}
+	mode := os.FileMode(perm)
+
+	writePath := filepath.Join(rootPath, file.Path)
```

Comment:
Can an attacker have file.Path include ".." to escape outside of the rootPath?
Using filepath.Clean on file.Path and then checking if the result starts with "
.." would capture such attempts.

Reply:
excellent catch!